src/document/position.js

```diff
@@ -97,13 +119,13 @@ CKEDITOR.define( [ 'utils', 'ckeditorerror' ], ( utils, CKEditorError ) => {
 		}
 
 		/**
-		 * Two positions equal if paths are equal.
+		 * Two positions equal if paths are equal and roots are the same.
 		 *
 		 * @param {document.Position} otherPosition Position to compare.
 		 * @returns {Boolean} True if positions equal.
 		 */
 		isEqual( otherPosition ) {
```

Comment:
Also to do `a.compareWith( b ) == Position.SAME` you need to require `Position` and you do not need to do it to do `a.isEqual( b )`. I agree that `isEqual` is faster to process in the brain, so we could add `Position.isBefore` and `Position.isAfter`.